src/data/real/sqrt.lean

```diff
@@ -0,0 +1,243 @@
+/-
+Copyright (c) 2020 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Floris van Doorn, Yury Kudryashov
+-/
+import topology.instances.nnreal
+
+/-!
+# Square root of a real number
+
+In this file we define
+
+* `nnreal.sqrt` to be the square root of a nonnegative real number.
+* `real.sqrt` to be the square root of a real number, defined to be zero on negative numbers.
+
+Then we prove some basic properties of these functions.
+
+## Implementation notes
+
+We define `nnreal.sqrt` as the noncomputable inverse to the function `x ↦ x * x`. We use general
+theory of inverses of strictly monotone functions to prove that `nnreal.sqrt x` exists. As a side
+effect, `nnreal.sqrt` is a bundled `order_iso`, so for `nnreal` numbers we get continuity as well as
```

Comment:
This is a little bit misleading, as you are not stating the continuity of the `nnreal` version below. Could you add it? (Maybe not all the versions with `continuous_on` and so on, but just the continuity would be nice).

Reply:
It's just `nnreal.sqrt.continuous`.

Reply:
I think it's worth adding it as a lemma, because people (like me :-) might not be aware that there is an automatic continuity statement for order isos.

Reply:
Two weeks ago it wasn't, see #5111. I'll add a lemma.